Replicator/tests/ReplicatorCollectionSGTest.cc

```diff
@@ -281,6 +365,125 @@ TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Multiple Collections Push & Pull S
     CHECK(_callbackStatus.progress.documentCount == 30 * _collectionCount);
 }
 
+
+TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Multiple Collections Push + Pull + (Push & Pull) SG",
+                 "[.SyncServerCollection]") {
+    string idPrefix = timePrefix();
+    const string channelID = idPrefix;
+    initTest({ Lavenders,
+               Roses,
+               Tulips },
+             { channelID }
+            );
+    enum {
+        iPush,
+        iPull,
+        iPushPull
+    };
+
+    constexpr slice body = "{\"ans*wer\":42}"_sl;
+    alloc_slice bodyWithChannel = SG::addChannelToJSON(body, "channels", {channelID});
+    alloc_slice localPrefix {idPrefix + "local-"};
+    alloc_slice remotePrefix {idPrefix + "remote-"};
+    unsigned docCount = 20;
+
+    // push documents to the pull and push/pull collections
+    for (size_t i = 0; i < _collectionCount; ++i) {
+        if (i == iPush) {
+            continue;
+        }
+        for (int d = 1; d <= docCount; ++d) {
+            constexpr size_t bufSize = 80;
+            char docID[bufSize];
+            snprintf(docID, bufSize, "%.*s%d", SPLAT(remotePrefix), d);
+            createFleeceRev(_collections[i], slice(docID), nullslice, bodyWithChannel);
+        }
+    }
+
+    {
+        // Send the docs to remote
+        ReplParams replParams { _collectionSpecs };
+        replParams.setPushPull(kC4OneShot, kC4Disabled);
+        replicate(replParams);
+    }
+
+    deleteAndRecreateDB();
+    initTest({Lavenders, Roses, Tulips}, {channelID});
```

Comment:
I suggest these two lines can be replaced with `deleteAndRecreateDBAndCollections()`. This will achieve the same outcome you're looking for, without the overhead of all the additional stuff in initTest.

Reply:
I will try that, thanks